docs/source/jit_language_reference_statements.rst

```diff
@@ -0,0 +1,389 @@
+::
+    for_stmt ::=  "for" target_list "in" expression_list ":" suite
+                  ["else" ":" suite]
+    `assignment_expression <https://docs.python.org/3/reference/expressions.html#grammar-token-assignment-expression>`
+    `expression_list <https://docs.python.org/3/reference/expressions.html#grammar-token-expression-list>`
+    `suite <https://docs.python.org/3/reference/compound_stmts.html#grammar-token-suite>`
+
+* `for...else` statements are not supported in Torchscript. It results in a `RuntimeError`
+
+* for loops on tuples: These unroll the loop,  generating a body for each member of the tuple.
+    * The body must type-check correctly for each member.
+
+    For eg.,
+    tup = (3, torch.rand(4))
+    for x in tup:
+        print(x)
+
+*  for loops on lists: For loops over a ``nn.ModuleList`` will unroll the body of the loop at compile time,
+    * with each member of the module list.
+
+    For eg.,
+
+    class SubModule(torch.nn.Module):
+        def __init__(self):
+            super(SubModule, self).__init__()
+            self.weight = nn.Parameter(torch.randn(2))
+
+        def forward(self, input):
+            return self.weight + input
+
+    class MyModule(torch.nn.Module):
+
+        def __init__(self):
+            super(MyModule, self).init()
+            self.mods = torch.nn.ModuleList([SubModule() for i in range(10)])
+
+        def forward(self, v):
+            for module in self.mods:
+                v = module(v)
+            return v
+
+    model = torch.jit.script(MyModule())
+
+The ``with`` statement:
+^^^^^^^^^^^^^^^^^^^^^^^
+* The `with <https://docs.python.org/3/reference/compound_stmts.html#with>` statement is used to wrap the
+* execution of a block with methods defined by a context manager
+
+::
+    with_stmt ::=  "with" with_item ("," with_item) ":" suite
+    with_item ::=  expression ["as" target]
+    `with_item <https://docs.python.org/3/reference/compound_stmts.html#grammar-token-with-item>`
+    `suite <https://docs.python.org/3/reference/compound_stmts.html#grammar-token-suite>`
+
+* If a target was included in the `with <https://docs.python.org/3/reference/compound_stmts.html#with>` statement,
+* the return value from context manager’s `__enter__() <https://docs.python.org/3/reference/datamodel.html#object.__enter__>`
+* is assigned to it. If an exception caused the suite to be exited, its type, value, and traceback are passed as
+* arguments to `__exit__() <https://docs.python.org/3/reference/datamodel.html#object.__exit__>. Otherwise, three
+* `None <https://docs.python.org/3/library/constants.html#None>` arguments are supplied.
+
+* `try/except/finally` statements are not supported inside `with` blocks.
+*  Exceptions raised within `with` block cannot be suppressed.
+
+The ``tuple`` statement:
+^^^^^^^^^^^^^^^^^^^^^^^
+
+::
+    tuple_stmt ::= tuple([iterables])
+
+* Iterable types in TorchScript include `Tensors`, `lists`, `tuples`, `dictionaries`, `strings`,
```

Comment:
Nit:
Use actual type names like how they are represented in Python: `dictionaries` vs `Dict`

Reply:
Done.